From 61e386ff293d4d4d62e3c14cf80b6f6889322e2d Mon Sep 17 00:00:00 2001
From: Tianyu Lan <tiala@microsoft.com>
Date: Wed, 13 Jul 2022 04:24:54 -0400
Subject: [PATCH 03/15] swiotlb: Check nslabs after initialization in the
 swiotlb_init_remap()

The check of nslabs less than IO_TLB_MIN_SLABS should be later than
initialization. Fix it.

Signed-off-by: Tianyu Lan <tiala@microsoft.com>
Link: https://lore.kernel.org/r/20220713082454.700487-1-ltykernel@gmail.com
---
 kernel/dma/swiotlb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/dma/swiotlb.c b/kernel/dma/swiotlb.c
index bf2ae98a42b4..8bd087cf0c53 100644
--- a/kernel/dma/swiotlb.c
+++ b/kernel/dma/swiotlb.c
@@ -302,9 +302,6 @@ void __init swiotlb_init_remap(bool addressing_limit, unsigned int flags,
 	if (swiotlb_force_disable)
 		return;
 
-	if (nslabs < IO_TLB_MIN_SLABS)
-		panic("%s: nslabs = %lu too small\n", __func__, nslabs);
-
 	/*
 	 * default_nslabs maybe changed when adjust area number.
 	 * So allocate bounce buffer after adjusting area number.
@@ -314,6 +311,9 @@ void __init swiotlb_init_remap(bool addressing_limit, unsigned int flags,
 
 	nslabs = default_nslabs;
 
+	if (nslabs < IO_TLB_MIN_SLABS)
+		panic("%s: nslabs = %lu too small\n", __func__, nslabs);
+
 	/*
 	 * By default allocate the bounce buffer memory from low memory, but
 	 * allow to pick a location everywhere for hypervisors with guest
-- 
2.37.1

